src/app/features/matching-entities/matching-entities/matching-entities.component.ts

```diff
@@ -0,0 +1,205 @@
+    this.rightFilters = config.rightFilters ?? this.rightFilters;
+    this.rightPrefilter = config.rightPrefilter ?? this.rightPrefilter;
+
+    if (!config.leftEntityType) {
+      this.leftEntitySelected = entity;
+    }
+    if (!config.rightEntityType) {
+      this.rightEntitySelected = entity;
+    }
+  }
+
+  private async init(side: "left" | "right") {
+    const sideIndex = side === "right" ? 1 : 0;
+
+    let entityType = this[side + "EntityType"];
+    if (typeof entityType === "string") {
+      entityType = this.entityRegistry.get(entityType);
+    }
+
+    const newSideDetails: MatchingSide = {
+      entityType: entityType ?? this[side + "EntitySelected"]?.getConstructor(),
+      selected: this[side + "EntitySelected"],
+      columns: this.columns?.map((p) => p[sideIndex]).filter((c) => !!c),
+      selectMatch: (e) => {
+        this.highlightSelectedRow(e, newSideDetails.selected);
+        newSideDetails.selected = e;
+        this.matchComparisonElement.nativeElement.scrollIntoView();
+      },
+    };
+
+    if (!newSideDetails.selected) {
+      newSideDetails.availableEntities = await this.entityMapper.loadType(
+        newSideDetails.entityType
+      );
+      newSideDetails.availableFilters = this[side + "Filters"];
+      newSideDetails.prefilter = this[side + "Prefilter"];
+    }
+
+    this.sideDetails[sideIndex] = newSideDetails;
+  }
+
+  private highlightSelectedRow(newSelectedEntity, previousSelectedEntity) {
+    if (previousSelectedEntity) {
+      previousSelectedEntity.getColor =
+        previousSelectedEntity.getConstructor().prototype.getColor;
+    }
+    newSelectedEntity.getColor = () =>
+      addAlphaToHexColor(newSelectedEntity.getConstructor().color, 0.2);
+  }
+
+  async createMatch() {
+    const newMatchEntity = new (this.entityRegistry.get(
+      this.onMatch.newEntityType
+    ))();
+    const selectedL = this.sideDetails[0].selected;
+    const selectedR = this.sideDetails[1].selected;
+
+    newMatchEntity[this.onMatch.newEntityMatchPropertyLeft] =
+      selectedL.getId(false);
+    newMatchEntity[this.onMatch.newEntityMatchPropertyRight] =
+      selectedR.getId(false);
+
+    // best guess properties (if they do not exist on the specific entity, the values will be discarded during save
+    newMatchEntity["date"] = new Date();
+    newMatchEntity["start"] = new Date();
+    newMatchEntity["name"] = `${
+      newMatchEntity.getConstructor().label
+    } ${selectedL.toString()} - ${selectedR.toString()}`;
+
+    if (this.onMatch.columnsToReview) {
+      this.formDialog.openSimpleForm(
+        newMatchEntity,
+        this.onMatch.columnsToReview
+      );
+    } else {
+      await this.entityMapper.save(newMatchEntity);
+    }
+
+    // lock in current selection to avoid duplicate matches and provide user feedback
+    this.sideDetails.forEach((s) => (s.availableEntities = undefined));
+  }
+
+  applySelectedFilters(side: MatchingSide, $event: DataFilter<Entity>) {
```

Comment:
```suggestion
  applySelectedFilters(side: MatchingSide, filter: DataFilter<Entity>) {
```
Let's not make this ambiguous. I am very worried that `$event` might be something internal (like it is in the template)